app/src/main/java/org/schabi/newpipe/database/Migrations.java

```diff
@@ -197,6 +198,47 @@ public void migrate(@NonNull final SupportSQLiteDatabase database) {
         }
     };
 
+    public static final Migration MIGRATION_6_7 = new Migration(DB_VER_6, DB_VER_7) {
+        @Override
+        public void migrate(@NonNull final SupportSQLiteDatabase database) {
+            // Create a new column thumbnail_stream_id
+            database.execSQL("ALTER TABLE `playlists` ADD COLUMN `thumbnail_stream_id` "
+                    + "INTEGER NOT NULL DEFAULT -1");
+
+            // Migrate the thumbnail_url to the thumbnail_stream_id
+            database.execSQL("CREATE TEMPORARY TABLE temporary_table AS"
+                    + " SELECT p.uid AS playlist_uid, s.uid AS stream_uid"
+                    + " FROM playlists p"
+                    + " LEFT JOIN playlist_stream_join ps ON p.uid = ps.playlist_id"
+                    + " LEFT JOIN streams s ON s.uid = ps.stream_id"
+                    + " WHERE s.thumbnail_url = p.thumbnail_url");
+
+            database.execSQL("UPDATE playlists SET thumbnail_stream_id = ("
+                    + "SELECT CASE WHEN COUNT(*) != 0 then stream_uid ELSE -1 END "
+                    + "FROM temporary_table "
```

Comment:
Why is the temporary table needed? Can't you replace `temporary_table` here with `(SELECT p.uid AS ...) AS temporary_table`?

Reply:
Yes, I used a temporary table because it is more readable. But I understand why you would want to replace it, so I changed it.